### PR TITLE
New eWallets API

### PR DESCRIPTION
**Pull request by glendaesutanto**
- `POST` Create eWallet charge
https://developers.xendit.co/api-reference/#create-ewallet-charge
- `GET` Get eWallet charge status
https://developers.xendit.co/api-reference/#get-ewallet-charge-status

### CODE REVIEW

File: xendit/models/ewallet/ewallet.py [hunk 1]
```diff
@@ -48,6 +54,45 @@ def helper_create_linkaja_item(*, id, name, price, quantity, **kwargs):
...
+          - name (str)
+          - category (str)
+          - currency (str)
+          - price (int)
+          - quantity (int)
+          - type (str)
+          - url (str)
+          - description (str)
+          - sub_category (str)
+          - metadata (dict)
```
Comment (weisurya) [inline xendit/models/ewallet/ewallet.py:86]: If the document states the data type for each parameter, can we set it on the function as well?
```suggestion
        *,
        reference_id: str,
        name: str,
        category: str,
        currency: str,
        price: int,
        quantity: int,
        type: str,
        url: str =None,
        description: str =None,
        sub_category: str =None,
        metadata:dict =None,
        **kwargs,
    ):
        """Construct Installments Object for Charge

        Args:
          - reference_id (str)
          - name (str)
          - category (str)
          - currency (str)
          - price (int)
          - quantity (int)
          - type (str)
          - url (str)
          - description (str)
          - sub_category (str)
          - metadata (dict)
```

File: xendit/models/ewallet/ewallet.py [hunk 2]
```diff
@@ -237,3 +282,90 @@ def _parse_ewallet_type(ewallet_type):
...
+          - currency (str)
+          - amount (int)
+          - checkout_method (str)
+          - channel_code (str)
+          - channel_properties (dict)
+          - customer_id (str)
+          - basket (list)
+          - metadata (dict)
+          - **for_user_id (str) (XenPlatforms only)
+          - **with_fee_rule (str) (XenPlatforms only)
```
Comment (weisurya) [inline xendit/models/ewallet/ewallet.py:315]: same argument like above
```suggestion
        *,
        reference_id: str,
        currency: str,
        amount: int,
        checkout_method: str,
        channel_code: str=None,
        channel_properties:dict =None,
        customer_id: str=None,
        basket:list =None,
        metadata: dict =None,
        for_user_id: str =None,
        with_fee_rule: str =None,
        **kwargs,
    ):
        """Send POST Request to create EWallet Charge (API Reference: eWallets/Create E-Wallet Charge)

        Args:
          - reference_id (str)
          - currency (str)
          - amount (int)
          - checkout_method (str)
          - channel_code (str)
          - channel_properties (dict)
          - customer_id (str)
          - basket (list)
          - metadata (dict)
          - **for_user_id (str) (XenPlatforms only)
          - **with_fee_rule (str) (XenPlatforms only)
```

File: xendit/models/ewallet/ewallet.py [hunk 3]
```diff
@@ -237,3 +282,90 @@ def _parse_ewallet_type(ewallet_type):
...
+        *,
+        charge_id,
+        for_user_id=None,
+        **kwargs,
+    ):
+        """Get ewallet charge status of given charge id (API Reference: eWallets/Get E-Wallet Charge Status)
+
+        Args:
+          - charge_id (str)
+          - **for_user_id (str) (XenPlatforms only)
```
Comment (weisurya) [inline xendit/models/ewallet/ewallet.py:350]: same comment like above
```suggestion
        *,
        charge_id: str,
        for_user_id: str =None,
        **kwargs,
    ):
        """Get ewallet charge status of given charge id (API Reference: eWallets/Get E-Wallet Charge Status)

        Args:
          - charge_id (str)
          - **for_user_id (str) (XenPlatforms only)
```

**Review by weisurya — COMMENTED** [review]
+ the linter test is failed

File: xendit/models/ewallet/ewallet.py (hunk 1 above)
Comment (glendaesutanto) [reply xendit/models/ewallet/ewallet.py:86]: Ooh I didn't know it could be like this, thanks for your review :+1:

**Comment by glendaesutanto** [comment]
> * the linter test is failed

@weisurya yes it's because credit card & virtual account integration test which is not implemented by me.. (for VA test, it passed in my local)

**Review by weisurya — APPROVED** [review]
After you test it and if it works (doesn't make any breaking changes), sounds good to me then

**Review by stanleynguyen — APPROVED** [review]

**Comment by dimasanwaraziz** [comment]
type object 'EWallet' has no attribute 'create_ewallet_charge'

**Pull request merged**